### PR TITLE
docs(x/ecocredit): update module overview

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #889

Update ecocredit overview and protobuf links.
 
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct `docs:` prefix in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the [documentation writing guidelines](https://github.com/cosmos/cosmos-sdk/blob/master/docs/DOC_WRITING_GUIDELINES.md)
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct `docs:` prefix in the PR title
- [ ] confirmed all author checklist items have been addressed 
- [ ] confirmed that this PR only changes documentation
- [ ] reviewed content for consistency
- [ ] reviewed content for thoroughness
- [ ] reviewed content for spelling and grammar
- [ ] tested instructions (if applicable)

### CODE REVIEW

File: x/ecocredit/spec/README.md [hunk 1]
```diff
@@ -1,10 +1,46 @@
...
+
+Credit classes are designed to support a variety of ecosystem service credits, and can be created and managed by individuals, groups, or organizations. Each credit class is associated with a credit type that represents the primary unit of measurement for the methodologies defined within the credit class (i.e. carbon, biodiversity, etc.). Credit types can only be added through on-chain governance.
+
+Ecosystem service credits are issued through a batch issuance process where only approved issuers for a given credit class can issue credits from that credit class. Each credit batch is associated with a project representing the on-the-ground project implementing the methodologies defined within the credit class. Credit batches are unique in that each credit batch has a start and end date representing the period in which the positive ecological outcomes were measured.
+
+Credits issued within a credit batch are only fungible with credits from the same batch. Credits can be issued in a retired or tradable state. Retiring a credit implies the credit is being consumed as an offset. Credits issued in a tradable state can be transferred or retired by the owner.
```
Comment (technicallyty) [inline x/ecocredit/spec/README.md:27]: i added a warning in the godoc of `Retire` that it's permanent. should we add that here?

**Review by technicallyty — APPROVED** [review]
looks good!

File: x/ecocredit/spec/README.md (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/spec/README.md:27]: Good idea. I can add.

Comment (ryanchristo) [reply x/ecocredit/spec/README.md]: https://github.com/regen-network/regen-ledger/pull/914/commits/42c66578370c03d2eb067a3775dc2457317f1e13

**Review by clevinson — APPROVED** [review]

**Pull request merged**